clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -157,4 +290,42 @@ public KafkaBasedCommandSender(
 
         return props;
     }
+
+    private Future<Void> subscribeForCommandResponse(final String tenantId,
+            final Handler<DownstreamMessage<KafkaMessageContext>> messageHandler, final Span span) {
+        if (commandResponseSubscriptions.get(tenantId) != null) {
+            span.log("command response consumer already exists");
+            LOGGER.debug("command response consumer already exists for tenant[{}]", tenantId);
+            return Future.succeededFuture();
+        } else {
+            final Map<String, String> consumerConfig = this.consumerConfig
+                    .getConsumerConfig(HonoTopic.Type.COMMAND_RESPONSE.toString());
+
+            // Use a unique group-id so that all command responses for this tenant are received by this consumer.
+            // Thereby the responses can be correlated with the command that has been sent.
+            consumerConfig.put(ConsumerConfig.GROUP_ID_CONFIG, tenantId + "-" + UUID.randomUUID().toString());
```

Comment:
Does this mean that there will be one consumer per in-flight command?

While I don't think this would be a serious issue for the Kafka broker itself, this would increase the network traffic quite a lot. With every in-flight command there would be a new consumer which will poll all currently present command responses.